### PR TITLE
fix(artifacts): set cache file permissions to read/write by all users (core)

### PR DESCRIPTION
**Pull request by moredatarequired**
Description
-----------

- Fixes [WB-20885](https://wandb.atlassian.net/browse/WB-20885)

The cache in `core` was intentionally restrictive with file permissions, but this doesn't account for all the ways users use it in production, in particular sharing the cache between multiple users. Expand the file permissions to allow read/write by all.

- [x] I updated CHANGELOG.md, or it's not applicable

Testing
-------
How was this PR tested?

[WB-20885]: https://wandb.atlassian.net/browse/WB-20885?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by ibindlish — APPROVED** [review]
lgtm

File: CHANGELOG.md [hunk 1]
```diff
@@ -11,6 +11,10 @@ Please add to the relevant subsections under Unreleased below on every PR where
 
 ## Unreleased
 
+### Fixed
+
+- Fixed file permissions for cache in core by @moredatarequired in https://github.com/wandb/wandb/pull/8362
```
Comment (timoffex) [inline CHANGELOG.md:16]: Mysterious!

> Fix `PermissionError` in `use_artifact` for artifacts downloaded with core enabled (@moredatarequired in ...)

is this correct?

**Review by timoffex — APPROVED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (moredatarequired) [reply CHANGELOG.md]: Better now?

Comment (timoffex) [reply CHANGELOG.md]: > Allow all users to read cache files when core is enabled

This should probably have the word "artifact" somewhere in there, and possibly mention what in the public API is affected. "cache files" seems like an artifact implementation detail.

File: CHANGELOG.md
Comment (timoffex) [inline CHANGELOG.md:1]: oops, looks like you had a merge conflict!

Comment (moredatarequired) [reply CHANGELOG.md:1]: I'm a little confused, I fixed this merge conflict in the GitHub UI and committed it from there--is there a footgun where it doesn't save changed properly?

Comment (timoffex) [reply CHANGELOG.md:1]: It looks like https://github.com/wandb/wandb/pull/8362/commits/a7956d4fa0ff0e71ae77c27e5bf693932767e63a committed a merge conflict. I haven't used GitHub's UI to resolve conflicts, so I'm not entirely sure how that can happen!

Comment (moredatarequired) [reply CHANGELOG.md:1]: https://github.com/wandb/wandb/pull/8401

**Pull request merged**